src/MagentoDataLayerPublishManager.ts

```diff
@@ -24,113 +22,108 @@ import {
 import { MagentoDataLayerBase } from "./MagentoDataLayerBase";
 
 export default class MagentoDataLayerPublishManager extends MagentoDataLayerBase {
-  constructor(mdl: MagentoDataLayer) {
-    super();
-    this.mdl = mdl;
-  }
-
   /**
    * Publish Add to Cart event
    */
-  addToCart(): void {
-    this.pushEvent(ADD_TO_CART);
+  addToCart(context?: Record<string, unknown>): void {
```

Comment:
Could we make all these `Record<string, unknown>` into the type`MagentoDataLayerContext` or is that string not necessarily one of the context names in src/types/contexts.ts?

```suggestion
  addToCart(context?: MagentoDataLayerContext): void {
```

Reply:
The context in the event publishers are meant to be optional custom contexts provided by the user. They may or may not be overriding one of the data layer contexts, so yeah it's purposely generic. I could create a `CustomContext` type for all of these Record<string, unknown> types if you want.